metrics/ci_integrations/test/cli/command/validate_command_test.dart

```diff
@@ -0,0 +1,532 @@
+        );
+
+        expect(
+          validateCommand.rawIntegrationConfigFactory,
+          equals(rawIntegrationConfigFactory),
+        );
+        expect(
+          validateCommand.configuredPartiesFactory,
+          equals(configuredPartiesFactory),
+        );
+        expect(
+          validateCommand.validationResultPrinter,
+          equals(validationResultPrinter),
+        );
+      },
+    );
+
+    test(
+      ".run() creates the raw integration config using the raw integration config factory and the config file path",
+      () {
+        whenCreateConfiguredParties().thenReturn(configuredParties);
+        whenValidateSourceConfig().thenAnswer(
+          (_) => Future.value(sourceValidationResult),
+        );
+        whenValidateDestinationConfig().thenAnswer(
+          (_) => Future.value(destinationValidationResult),
+        );
+
+        commandStub.run();
+
+        verify(rawIntegrationConfigFactory.create(configFilePath)).called(1);
+      },
+    );
+
+    test(
+      ".run() throws a ConfigValidationError if there is an error occurred during the raw integration config creation",
+      () {
+        when(
+          rawIntegrationConfigFactory.create(configFilePath),
+        ).thenThrow(Exception());
+
+        expect(() => commandStub.run(), throwsConfigValidationError);
+      },
+    );
+
+    test(
+      ".run() creates the configured parties using the configured parties factory and the raw integration config",
+      () {
+        whenCreateConfiguredParties().thenReturn(configuredParties);
+        whenValidateSourceConfig().thenAnswer(
+          (_) => Future.value(sourceValidationResult),
+        );
+        whenValidateDestinationConfig().thenAnswer(
+          (_) => Future.value(destinationValidationResult),
+        );
+
+        commandStub.run();
+
+        verify(configuredPartiesFactory.create(rawIntegrationConfig)).called(1);
+      },
+    );
+
+    test(
+      ".run() throws a ConfigValidationError if there is an error occurred during the configured parties creation",
+      () {
+        when(
+          rawIntegrationConfigFactory.create(configFilePath),
+        ).thenReturn(rawIntegrationConfig);
+        when(
+          configuredPartiesFactory.create(rawIntegrationConfig),
+        ).thenThrow(Exception());
+
+        expect(() => commandStub.run(), throwsConfigValidationError);
+      },
+    );
+
+    test(
+      ".run() creates the source config validator using the config validator factory and the config from the configured source party",
+      () {
+        whenCreateConfiguredParties().thenReturn(configuredParties);
+        whenValidateSourceConfig().thenAnswer(
+          (_) => Future.value(sourceValidationResult),
+        );
+        whenValidateDestinationConfig().thenAnswer(
+          (_) => Future.value(destinationValidationResult),
+        );
+
+        commandStub.run();
+
+        verify(sourceConfigValidatorFactory.create(sourceConfig)).called(1);
```

Comment:
```suggestion
        final sourceConfig = configuredParties.configuredSourceParty.config;

        commandStub.run();

        verify(sourceConfigValidatorFactory.create(sourceConfig)).called(1);
```
Would it make tests more readable?